wrappers/windres-wrapper.sh

```diff
@@ -1,3 +1,183 @@
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# For more information, please refer to <http://unlicense.org/>
+
+PROG="$0"
+
+CC=${CC:-i686-w64-mingw32-clang}
+
+print_version () {
+    cat <<EOF >&2
+version: ?xd
+EOF
+    exit 1
+}
+
+print_help () {
+ cat <<EOF >&2
+usage: llvm-windres <OPTION> [INPUT-FILE] [OUTPUT-FILE]
+
+LLVM Tool to manipulate Windows resources with a GNU windres interface.
+
+Options:
+  -i, --input <arg>          Name of the input file.
+  -o, --output <arg>         Name of the output file.
+  -J, --input-format <arg>   Input format to read.
+  -O, --output-format <arg>  Output format to generate.
+  --preprocessor <arg>       Custom preprocessor command.
+  --preprocessor-arg <arg>   Preprocessor command arguments.
+  -F, --target <arg>         Target for COFF objects to be compiled for.
+  -I, --include-dir <arg>    Include directory to pass to preprocessor and resource compiler.
+  -D, --define <arg[=val]>   Define to pass to preprocessor.
+  -U, --undefine <arg[=val]> Undefine to pass to preprocessor.
+  -c, --codepage <arg>       Default codepage to use when reading an rc file (0x0-0xffff).
+  -v, --verbose              Enable verbose output.
+  -V, --version              Display version.
+  -h, --help                 Display this message and exit.
+Input Formats:
+  rc                         Text Windows Resource
+  res                        Binary Windows Resource
+Output Formats:
+  res                        Binary Windows Resource
+  coff                       COFF object
+Targets:
+  armv7-w64-mingw32
+  aarch64-w64-mingw32
+  i686-w64-mingw32
+  x86_86-w64-mingw
+EOF
+    exit 1
+}
+
+error() {
+    echo "${PROG}: error: $1" >&2
+    exit 1
+}
+
+quote() {
+    echo "$1" | sed -e "s|'|'\\\\''|g; 1s/^/'/; \$s/\$/'/"
+}
+
+INCLUDE=
+VERBOSE=${VERBOSE:-false}
+INPUT=-
+OUTPUT=/dev/stdout
+INPUT_FORMAT=rc
+OUTPUT_FORMAT=res
+TARGET=X64
```

Comment:
This default value here doesn't seem to match the namespace of target values described in the help.

Also, you should probably try to look at `$0` and see if we should set a specific default target arch based on what the tool is called. In dlltool-wrapper.sh, I'm doing this:
```
TARGET="$(basename $0 | sed 's/-[^-]*$//')"
ARCH=$(echo $TARGET | sed 's/-.*//')
```
(For these cases it could be simplified to just a single sed call as well, instead of parsing out the full target string inbetween.)

I guess it'd be even better to account for the case when the tool is invoked without such a matching prefix at all as well. Or if it's only ever going to be used via `<triplet>-windres` symlinks it probably doesn't matter.

Reply:
I was wondering how to solve this, assuming it will be called `<triplet>-windres` is probably fine.